bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/AutoUpdateManager.java

```diff
@@ -174,27 +176,41 @@ public void receiveCommand(ItemCommandEvent commandEvent, Item item) {
     private Recommendation shouldAutoUpdate(String itemName) {
         Recommendation ret = Recommendation.REQUIRED;
 
-        Set<ChannelUID> linkedChannelUIDs = itemChannelLinkRegistry.stream().filter(link -> {
-            return link.getItemName().equals(itemName);
-        }).map(link -> {
-            return link.getLinkedUID();
-        }).collect(toSet());
+        List<ChannelUID> linkedChannelUIDs = new ArrayList<>();
```

Comment:
No need to change, but imho the stream was also readable

Reply:
The stream caused the tests to be not reproducible as it didn't retain the order.